src/main/java/com/example/petbuddybackend/entity/user/Caretaker.java

```diff
@@ -15,21 +17,28 @@
 public class Caretaker {
 
     @Id
-    private String email;
+    private Long id;
 
     @Column(nullable = false, length = 14)
     private String phoneNumber;
 
     @Column(length = 1500)
     private String description;
 
-    @OneToMany(cascade = CascadeType.ALL, mappedBy = "caretaker")
-    private List<AnimalTakenCareOf> animalsTakenCareOf;
-
-    @OneToOne(cascade = CascadeType.ALL, optional = false)
-    private Address address;
-
     @PrimaryKeyJoinColumn
     @OneToOne(cascade = CascadeType.MERGE, optional = false)
     private AppUser accountData;
+
+    @Basic(fetch = FetchType.LAZY)
+    @Formula("(SELECT AVG(r.rating) FROM Rating r WHERE r.caretaker_id = id)")
```

Comment:
dlaczego tutaj jest caretaker_id zamiast caretakerId?

Reply:
To jest zwykłe zapytanie niezwiązane z JPA. W bazie mamy snake case nazw atrybutów więc tutaj inna konwencja nie działa